File: testros2/yuxiangros/chapter_2/charpter_workspace/src/demo_python_pkg/demo_python_pkg/person_node.py
class PersonNode:
    def __init__(self, name_value: str, age_value: int)->None:
        print("PersonNode __init__方法被调用了，添加了两个属性")
        self.name = name_value
        self.age = age_value

    def eat(self, food:str)->None:
        """
        吃饭--方法
        """
        print("{},{}岁，爱吃{}".format(self.name, self.age, food))

def main():
    node = PersonNode("陈超", 18)
    node.eat("beef, fish, meat")

if __name__ == "__main__":
    main()

File: testros2/yuxiangros/chapter_2/charpter_workspace/src/demo_cpp_pkg/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode
:public rclcpp::Node
{
public:
    //构造函数
    PersonNode(const std::string& nodeName, const std::string& name, const int& age)
    :Node(nodeName),
    name_(name),
    age_(age)
    {
        RCLCPP_INFO(this->get_logger(), "%s: Person Node has been created.", __func__);
        RCLCPP_INFO(this->get_logger(), "Name: %s, age: %d", name_.c_str(), age_);
    }
    //析构函数
    ~PersonNode()
    {
        RCLCPP_INFO(this->get_logger(), "Person Node has been destroyed.");   
    }
public:
    void eat(const std::string foodName)
    {
        RCLCPP_INFO(this->get_logger(), "%s is eating %s.", name_.c_str(), foodName.c_str());
    }
private:
    std::string name_;
    int age_;
};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("PersonNode", "chenchao", 30);
    node->eat("apple");
    RCLCPP_INFO(node->get_logger(), "Hello, 30!");
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}